### PR TITLE
User details update form: 

### PR DESCRIPTION
**Pull request by tulitheprogrammer**
Enter key fixed to submit the form instead of clearing the text input field.
Workaround added for HTML forms natural behaviour to press first form button on Enter key press
#190

### CODE REVIEW

File: src/app/pages/user/dialogs/update/update-user.dialog.html [hunk 1]
```diff
@@ -56,17 +56,16 @@ <h3 mat-dialog-title>עדכון פרטים אישיים</h3>
...
-    <div mat-dialog-actions>
-      <button  mat-button class="msw-button-cancel" tabindex="-1" (click)="close()">ביטול</button>
-      <!-- <button [type]="submit" class="mat-raised-button mat-primary" (click)="updateDetails()" [disabled]="!formControl.valid" >עדכן משתמש</button> -->
-      <button  mat-button class="msw-button-update" (click)="updateDetails()" [disabled]="!formControl.valid" >עדכן</button>
-    </div>
+      </mat-dialog-content>
+      <div mat-dialog-actions>
+        <button  mat-button class="msw-button-cancel" tabindex="-1" (click)="close()">ביטול</button>
+         <!-- <button [type]="submit" class="mat-raised-button mat-primary" (click)="updateDetails()" [disabled]="!formControl.valid" >עדכן משתמש</button> -->
+        <button type="submit" mat-button class="msw-button-update" [disabled]="!formControl.valid" >עדכן</button>
```
Comment (eran1205) [inline src/app/pages/user/dialogs/update/update-user.dialog.html]: (click)="updateDetails()"  is missing in that line. How do the button know what function to activate?

File: src/app/pages/user/dialogs/update/update-user.dialog.html [hunk 2]
```diff
@@ -56,17 +56,16 @@ <h3 mat-dialog-title>עדכון פרטים אישיים</h3>
...
-    </mat-dialog-content>
-    <div mat-dialog-actions>
-      <button  mat-button class="msw-button-cancel" tabindex="-1" (click)="close()">ביטול</button>
-      <!-- <button [type]="submit" class="mat-raised-button mat-primary" (click)="updateDetails()" [disabled]="!formControl.valid" >עדכן משתמש</button> -->
-      <button  mat-button class="msw-button-update" (click)="updateDetails()" [disabled]="!formControl.valid" >עדכן</button>
-    </div>
+      </mat-dialog-content>
+      <div mat-dialog-actions>
+        <button  mat-button class="msw-button-cancel" tabindex="-1" (click)="close()">ביטול</button>
+         <!-- <button [type]="submit" class="mat-raised-button mat-primary" (click)="updateDetails()" [disabled]="!formControl.valid" >עדכן משתמש</button> -->
```
Comment (dimkinv) [inline src/app/pages/user/dialogs/update/update-user.dialog.html]: I see this comment was before but if you fixed the issue around it, can you please remove the comment as well?

Thanks.

File: src/app/pages/user/dialogs/update/update-user.dialog.html (hunk 1 above)
Comment (tulitheprogrammer) [reply src/app/pages/user/dialogs/update/update-user.dialog.html]: Hi @eran1205,
This button is now the form's "official" submit button.
Both Enter keypress and this button click triggers the "updateDetails()" handler

File: src/app/pages/user/dialogs/update/update-user.dialog.html (hunk 2 above)
Comment (tulitheprogrammer) [reply src/app/pages/user/dialogs/update/update-user.dialog.html]: Sure thing.
Done

**Pull request merged**